tests/assembleCoaddTestUtils.py

```diff
@@ -479,7 +443,11 @@ def makeDataRefList(exposures, matchedExposures, warpType, tract=0, patch=42, co
                 exposure = matchedExposures[expId]
             else:
                 raise ValueError("warpType must be one of 'direct' or 'psfMatched'")
-            dataRef = MockWarpReference(exposure, coaddName=coaddName,
-                                        tract=tract, patch=patch, visit=expId)
+            dataId = lsst.daf.butler.DataCoordinate.standardize(
+                tract=tract, patch=patch, visit=expId,
+                instrument="DummyCam", skymap="Skymap",
+                universe=lsst.daf.butler.DimensionUniverse(),
+            )
+            dataRef = MockWarpReference(exposure, dataId=dataId)
```

Comment:
Is the preferred pattern this (hand the dataId to the mock initializer) or what is done below (override init and construct the dataId in the mock initializer)?  I think we should pick a pattern and stick with it for clarity now and in the future.

Reply:
If the `InMemoryDataset` had something like `mock_dataid=dict()` that then turned some arbitrary dict into a generic dataid, wouldn't that work generally for these sorts of purposes, and also be clear that this was just for mocking in tests?

Reply:
The `dataId=` for an `InMemoryDatasetHandle` is meant to be the mock dataId already -- there is no need to add an alternative mock dataId. Allowing it to be a `dict` but turning it into some kind of mocked DataCoordinate internally should be okay (we actually have a mocked DataCoordinate thing in butler testing).

Reply:
Is this needed now, or can it accept a dict with recent updates?

Reply:
Changed it, and now also retain the coaddName since it no longer has to be a real DataCoordinate.